test/support/single_cov.rb

```diff
@@ -0,0 +1,67 @@
+class SingleCov
+  COVERAGES = []
+
+  class << self
+    def expect(file, percent)
+      COVERAGES << [file, percent]
+    end
+
+    def verify!(result)
+      COVERAGES.each do |file, expected_percent|
+        uncovered!(file) unless coverage = result[File.expand_path(file)]
+
+        line_count = coverage.count { |l| l }
+        covered_lines = coverage.count { |l| l && l > 0 }
+        actual_percent = 100 * covered_lines / line_count
+        next if actual_percent == expected_percent
+
+        details = "#{actual_percent}% vs #{expected_percent}%"
+        if actual_percent > expected_percent
+          warn "#{file} exceeds expected coverage #{details}, increment expected coverage?"
+        else
+          warn "#{file} lower then expected coverage #{details}"
+          warn "Lines missing coverage:"
+          warn coverage.select { |c| c == 0 }.each_with_index.map { |c, i| "#{file}:#{i+1}" }
+          exit 1
```

Comment:
this doesn't actually kill the process immediately because everything is running `at_exit`? It just makes sure the code is not 0?